src/app/routing/components/AppRoute.tsx

```diff
@@ -1,26 +1,28 @@
-import PropTypes from "prop-types";
 import React, { useState } from "react";
 import { useSelector } from "react-redux";
-import { Redirect, Route } from "react-router-dom";
+import { Redirect, Route, RouteProps, RouteChildrenProps } from "react-router-dom";
 import { isLoaded } from "react-redux-firebase";
 import { makeStyles } from "@material-ui/core/styles";
 import LinearProgress from "@material-ui/core/LinearProgress";
 
 import RoutingService from "../services/RoutingService";
-import { UserPermissionsService } from "../../model/permissions";
+import { UserPermissionsService, USER_ROLES } from "../../model/permissions";
 
 const loadingStyles = makeStyles((theme) => ({
   root: {
     width: "100%",
   },
 }));
 
-function AppRoute({ children, component, ...rest }) {
+type Props = RouteProps;
+
+function AppRoute({ children, component, ...rest }: Props) {
```

Comment:
@amylo I converted this component to typescript. lemme know if I'm missing anything